### PR TITLE
Bug 1877860: Volume Snapshot basic CRUD tests

### PR DESCRIPTION
**Pull request by cloudbehl**
Signed-off-by: Ankush Behl <cloudbehl@gmail.com>

### CODE REVIEW

**Comment by dtaylor113** [comment]
Hi @cloudbehl are this tests in addition too, and will they impact/conflict with:  #6340 ?

**Comment by bipuladh** [comment]
/hold
We need to separate the tests from the Bug fixes.
https://github.com/openshift/console/pull/6340 also handles CRUD tests of Volume Snapshot already.

**Comment by cloudbehl** [comment]
@dtaylor113 Yes these are additional test which helps generic flows like YAML based creation, edit resource & deletion of the resource.

@bipuladh canceling hold as discussed. The bug is related to tests where YAML creation button was not present & the tests were not getting passed.

/hold cancel

**Comment by dtaylor113** [comment]
Hi @cloudbehl, irt. the changes to Cypress integration test changes in `k8-openshift-cruds.spec.ts`, those LGTM.   I'm not familiar with the other Volume Snapshot changes made.

File: frontend/public/models/yaml-templates.ts [hunk 1]
```diff
@@ -1210,6 +1210,48 @@ spec:
...
+deletionPolicy: Delete
+`,
+  )
+  .setIn(
+    [referenceForModel(k8sModels.VolumeSnapshotContentModel), 'default'],
+    `
+apiVersion: snapshot.storage.k8s.io/v1beta1
+kind: VolumeSnapshotContent
+metadata:
+  name: new-snapshot-content-test
```
Comment (spadgett) [inline frontend/public/models/yaml-templates.ts:1245]: Should we have "test" in the name if this is going to be the real volume snapshot template? (Here are elsewhere in the samples.)

We try to make the examples just work if you click save with no changes, but I imagine that's not possible here.

File: frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx [hunk 2]
```diff
@@ -201,84 +210,97 @@ const CreateSnapshotForm = withHandlePromise<SnapshotResourceProps>((props) => {
...
-        </ButtonBar>
-      </form>
+          {pvcObj && (
+            <div className="form-group co-volume-snapshot__form">
+              <label className="control-label co-required" htmlFor="snapshot-class">
+                Snapshot Class
+              </label>
+              <SnapshotClassDropdown
+                onChange={setSnapshotClassName}
+                id="claimName"
```
Comment (spadgett) [inline frontend/packages/console-app/src/components/volume-snapshot/create-volume-snapshot/create-volume-snapshot.tsx:281]: Duplicate ID.

File: frontend/public/models/yaml-templates.ts (hunk 1 above)
Comment (cloudbehl) [reply frontend/public/models/yaml-templates.ts:1245]: > Should we have "test" in the name if this is going to be the real volume snapshot template? (Here are elsewhere in the samples.)
Right, we can call it an example/test. I'll change it.
> We try to make the examples just work if you click save with no changes, but I imagine that's not possible here.
I agree, Right now volume snapshots are not adapted to all the storage providers as it's not GA'ed. When snapshot gets GA'ed we should get the default CSI hostpath driver out of the box. So for now just added an example of it as this is provided on k8s documentation as well.

**Comment by cloudbehl** [comment]
@spadgett updated the PR please have a look.

**Review by spadgett — APPROVED** [review]
/lgtm

**Comment by cloudbehl** [comment]
/hold cancel
Rebased the PR!

**Review by dtaylor113 — APPROVED** [review]
/lgtm -thanks

**Review by spadgett — APPROVED** [review]
/approve

**Pull request merged**